plugin/storage/grpc/factory.go

```diff
@@ -63,14 +63,15 @@ func NewFactory() *Factory {
 // NewFactoryWithConfig is used from jaeger(v2).
 func NewFactoryWithConfig(
 	cfg Config,
-	metricsFactory metrics.Factory,
-	logger *zap.Logger,
 	host component.Host,
+	metricsFactory metrics.Factory,
+	telset component.TelemetrySettings,
```

Comment:
I meant Jaeger's internal `telemetry.Settings`, which includes both host and metrics factory

Reply:
`telset := telemetery.Setting{
				Logger: s.telset.Logger,
				Host: host,
				Metrics: mf,
				LeveledMeterProvider: func(_ configtelemetry.Level)  metric.MeterProvider {
					return s.telset.MeterProvider
				 },
			}`

Reply:
can i pass meter provider like this  ? or should i have a field to pass meter provider in function signature ?

Reply:
yes that's fine (ie to ignore the level)